src/python/pants/engine/internals/specs_rules.py

```diff
@@ -575,9 +574,7 @@ async def find_valid_field_sets_for_target_roots(
             result,
             cast(
```

Comment:
We should be able to remove a lot of the ambiguity handling code now that we aren't playing funny games with ambiguous python specs.

Basically, it shouldn't need the field set type anymore or be trying to decide if we should choose one target or another. Ambiguity is now just ambiguity.

Reply:
Not too well versed in this area, so would appreciate your help there as well.

Reply:
Yeah no worries. I'll try to take a stab tonight or tomorrow. 👍

Reply:
pushed :raised_hands: